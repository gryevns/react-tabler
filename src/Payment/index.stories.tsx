import React from "react";

import Payment from "../Payment";

export default {
    title: "Payment",
};

export const All = () => {
    const context = require.context("../static/img/payments", true, /.svg$/);

    const providers = context
        .keys()
        .map(key => {
            return key.split("./")[1].split(".svg")[0];
        })
        .filter(provider => !provider.includes("-dark"));

    return providers.map(provider => <Payment provider={provider} size="xl" />);
};

export const Sizes = () =>
    [undefined, "lg" as const, "xl" as const].map(size => (
        <Payment provider="paypal" size={size} />
    ));
