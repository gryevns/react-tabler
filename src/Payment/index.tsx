import React from "react";
import cn from "classnames";

export type Size = "lg" | "xl";

export interface PaymentProps {
    className?: string;
    provider: string;
    size?: Size;
}

const Payment = ({ className, provider, size }: PaymentProps) => {
    const classes = cn(
        "payment payment-sm",
        { [`payment-provider-${provider}`]: provider, [`payment-${size}`]: !!size },
        className,
    );
    return <span className={classes}></span>;
};

export default Payment;
