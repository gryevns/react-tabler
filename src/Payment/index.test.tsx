import React from "react";
import { shallow } from "enzyme";

import Payment from "./index";

describe("Payment", () => {
    it("should match snapshot", () => {
        [undefined, "lg" as const, "xl" as const].map(size => {
            const wrapper = shallow(<Payment provider="paypal" size={size} />);
            expect(wrapper).toMatchSnapshot();
        });
    });
});
