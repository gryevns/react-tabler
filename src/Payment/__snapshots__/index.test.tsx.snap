// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment should match snapshot 1`] = `
<span
  className="payment payment-sm payment-provider-paypal"
/>
`;

exports[`Payment should match snapshot 2`] = `
<span
  className="payment payment-sm payment-provider-paypal payment-lg"
/>
`;

exports[`Payment should match snapshot 3`] = `
<span
  className="payment payment-sm payment-provider-paypal payment-xl"
/>
`;
